include/sparrow/buffer/allocator.hpp

```diff
@@ -111,12 +111,19 @@ namespace sparrow
 
             [[nodiscard]] T* allocate(std::size_t n) override
             {
-                return m_alloc.allocate(n);
+                return reinterpret_cast<T*>(m_alloc.allocate(n));
```

Comment:
This `reinterpret_cast` should not be required, we should ensure that `T` and `A::value_type` are the same in the `any_allocator` constructor. This would also allow to remove the reinterpret cast in the deallocate, and probably remove the surrounding pragma.

I understand this is required so that the current implementation of the u8_buffer can take owneship of an already allocated buffer of a different type, and I don't want to delay this too much, but I think this should be refactored with a specific allocator adaptor in the near future (so maybe just add a TODO when you restore the GCC pragmas).